lib/modules/datasource/crate/index.ts

```diff
@@ -371,4 +377,39 @@ export class CrateDatasource extends Datasource {
 
     return [packageName.slice(0, 2), packageName.slice(2, 4), packageName];
   }
+
+  private static releaseTimestampSchema = z
+    .object({
+      version: z.object({
+        created_at: z.string(),
+      }),
+    })
+    .transform(({ version: { created_at } }) => created_at);
+
+  @cache({
+    namespace: `datasource-crate`,
+    key: (
+      { registryUrl, packageName }: PostprocessReleaseConfig,
+      { version }: Release,
+    ) => `postprocessRelease:${registryUrl}:${packageName}:${version}`,
+    ttlMinutes: 24 * 60,
+    cacheable: ({ registryUrl }: PostprocessReleaseConfig, _: Release) =>
+      registryUrl === 'https://crates.io',
+  })
+  async postprocessRelease(
+    { packageName, registryUrl }: PostprocessReleaseConfig,
+    release: Release,
+  ): Promise<Release | null> {
+    if (registryUrl !== 'https://crates.io') {
+      return release;
+    }
+
+    const url = `https://crates.io/api/v1/crates/${packageName}/${release.version}`;
+    const { body: releaseTimestamp } = await this.http.getJson(
+      url,
+      CrateDatasource.releaseTimestampSchema,
+    );
```

Comment:
Does this need a try/catch for safety?

Reply:
I think we're okay with standard error handling, which just returns the release object intact

Reply:
Changed schema to be nullable